### PR TITLE
Update html file comparison in e2e tests

### PR DESCRIPTION
**Pull request by chejennifer**
- When generating charts for the summary report, depending on the machine, the dimensions may vary slightly. In the e2e tests, instead of comparing the actual vs expected summary_report.html files just as strings, compare the elements on the page and their text content. This tests the important parts of the html file

### CODE REVIEW

File: tool/src/test/java/org/datacommons/tool/LintTest.java [hunk 1]
```diff
@@ -71,8 +70,8 @@ public void LintTest() throws IOException {
           if (f.equals("report.json")) {
             TestUtil.assertReportFilesAreSimilar(
                 expect, TestUtil.readStringFromPath(expected), TestUtil.readStringFromPath(actual));
-          } else {
-            assertEquals(
+          } else if (f.equals("summary_report.html")) {
+            TestUtil.assertHtmlFilesAreSimilar(
                 TestUtil.readStringFromPath(expected), TestUtil.readStringFromPath(actual));
           }
```
Comment (pradh) [inline tool/src/test/java/org/datacommons/tool/LintTest.java:76]: To confirm, there's no else case here?

File: tool/src/test/java/org/datacommons/tool/TestUtil.java [hunk 2]
```diff
@@ -68,4 +74,28 @@ public static String readStringFromPath(Path filePath) throws IOException {
...
+      Element expectedElement = expectedElements.poll();
+      Element actualElement = actualElements.poll();
+      expectedElements.addAll(expectedElement.children());
+      actualElements.addAll(actualElement.children());
+      String expectedElementText = expectedElement.ownText();
+      String actualElementText = actualElement.ownText();
+      assertEquals(
+          "Actual HTML file has an element that differs in text content from what is expected",
+          expectedElementText,
+          actualElementText);
```
Comment (pradh) [inline tool/src/test/java/org/datacommons/tool/TestUtil.java:94]: Is it worth using expect so we don't stop on the first diff?

File: tool/src/test/java/org/datacommons/tool/TestUtil.java [hunk 3]
```diff
@@ -68,4 +74,28 @@ public static String readStringFromPath(Path filePath) throws IOException {
...
+    }
+    assertTrue(
```
Comment (pradh) [inline tool/src/test/java/org/datacommons/tool/TestUtil.java:96]: Print the elements to aid debugging?

File: tool/src/test/java/org/datacommons/tool/TestUtil.java [hunk 4]
```diff
@@ -68,4 +74,28 @@ public static String readStringFromPath(Path filePath) throws IOException {
...
+    Parser htmlParser = Parser.htmlParser();
+    Document expectedHtml = htmlParser.parseInput(expected, "expected");
+    Document actualHtml = htmlParser.parseInput(actual, "actual");
+    LinkedList<Element> expectedElements = new LinkedList<>(expectedHtml.children());
+    LinkedList<Element> actualElements = new LinkedList<>(actualHtml.children());
+    while (!expectedElements.isEmpty() && !actualElements.isEmpty()) {
+      Element expectedElement = expectedElements.poll();
+      Element actualElement = actualElements.poll();
+      expectedElements.addAll(expectedElement.children());
+      actualElements.addAll(actualElement.children());
```
Comment (pradh) [inline tool/src/test/java/org/datacommons/tool/TestUtil.java:88]: I wonder about append all the actual and expected ownTexts into a single long string and comparing them?  Would the output of that be more readable than doing piecemeal (e.g., first element is an addition, there are extra elements, etc)?

File: tool/src/test/java/org/datacommons/tool/TestUtil.java (hunk 2 above)
Comment (pradh) [reply tool/src/test/java/org/datacommons/tool/TestUtil.java:94]: (Moot if we do a single full text assert)

File: tool/src/test/java/org/datacommons/tool/TestUtil.java (hunk 3 above)
Comment (pradh) [reply tool/src/test/java/org/datacommons/tool/TestUtil.java:96]: (Moot if we do a single full text assert)

File: tool/src/test/java/org/datacommons/tool/LintTest.java (hunk 1 above)
Comment (chejennifer) [reply tool/src/test/java/org/datacommons/tool/LintTest.java:76]: yes, no else case here

File: tool/src/test/java/org/datacommons/tool/TestUtil.java (hunk 4 above)
Comment (chejennifer) [reply tool/src/test/java/org/datacommons/tool/TestUtil.java:90]: I like that idea, updated!

Comment (pradh) [reply tool/src/test/java/org/datacommons/tool/TestUtil.java:90]: Thanks. Maybe we can append *all* expectedElements and actualElements and then compare them?

Then we can get rid of the two assertTrue() statements below?

(Feel free to submit :) )

**Review by pradh — APPROVED** [review]
Thanks for the change!

**Pull request merged**